src/development/vortex/development/core/engine/trainer/base_trainer.py

```diff
@@ -126,7 +114,35 @@ def create_scheduler(scheduler, optimizer):
                 "unsupported lr_scheduler {}".format(sch_method)
             kwargs.update({'optimizer': optimizer})
             scheduler = getattr(lr_scheduler, sch_method)(**kwargs)
+
+            # Assign scheduler type update type
+            scheduler.step_update = None
+            for step_update_type in lr_scheduler.step_update_map:
+                if type(scheduler).__name__ in lr_scheduler.step_update_map[step_update_type]:
+                    scheduler.step_update = step_update_type
+            if scheduler.step_update == None:
+                raise RuntimeError('Currently, scheduler {} is not supported, please select other scheduler'.format(type(scheduler).__name__))
         return scheduler
```

Comment:
there is no more checks for type other than `dict` for `scheduler`?

Reply:
no currently, and I only remove the deprecation warning

Reply:
would probably be better if it also be checked, but I think it is fine for now.